### ISSUE TITLE
Add support for List customer payments

### ISSUE THREAD

**Issue by pGuernseyPlayer** [Viincenttt/MollieApi#94]
Mollie's function for List customer payments; Customers API v2
is described here
https://docs.mollie.com/reference/v2/customers-api/list-customer-payments

I added this method to the ICustomerClient interface:'
Task<ListResponse<PaymentResponse>> GetCustomerPaymentListAsync(UrlObjectLink<ListResponse<PaymentResponse>> url);

The solution at [https://github.com/Viincenttt/MollieApi/issues/86] does not apply, unless I missed something.
I implemented it in the CustomerClient class:

        public async Task<ListResponse<PaymentResponse>> GetCustomerPaymentListAsync(UrlObjectLink<ListResponse<PaymentResponse>> url)
        {
            // should be:
            // https://api.mollie.com/v2/customers/cst_h929p86SHJ/payments

            return await this.GetListAsync<ListResponse<PaymentResponse>>(fullUrl: url.Href).ConfigureAwait(false);
        }

EXAMPLE of calling the above implementation:
            ListResponse<PaymentResponse> paymentList 
                = await customerClient.GetCustomerPaymentListAsync(mollieCust.Links.Payments);

and added this overload to the BaseMollieClient class:

        protected async Task<T> GetListAsync<T>(string fullUrl)
        {
            //var queryString = this.BuildListQueryString(from, limit, otherParameters);
            HttpRequestMessage httpRequest = this.CreateHttpRequest(HttpMethod.Get, fullUrl);
            var response = await this._httpClient.SendAsync(httpRequest).ConfigureAwait(false);
            return await this.ProcessHttpResponseMessage<T>(response).ConfigureAwait(false);
        }

HOWEVER, the ListResponseConverter.ReadJson method throws an exception : 
Newtonsoft.Json.JsonReaderException: Error reading string. Unexpected token: StartObject. Path 'payments[0].metadata',

**Comment by pGuernseyPlayer** [comment]
Ah, I just discovered that I basically re-wrote the PaymentClient.GetPaymentListAsync(UrlObjectLink<ListResponse<PaymentResponse>> url) 
method, but the same exception is encountered at the same line in ListResponseConverter.ReadJson();

I think this issue should be renamed, Exception deserializing PaymentResponse Json which contains metadata.

**Comment by pGuernseyPlayer** [comment]
Fixed by adding the attribute to the property in the PaymentResponse class:

        [JsonConverter(typeof(RawJsonConverter))]
        public string Metadata { get; set; }

**Comment by Viincenttt** [comment]
Hi pGuernseyPlayer,

Could you let me know what the Metadata value was when this exception occured? I haven't been able to reproduce this bug as of yet. 

Keep in mind that the [JsonConverter(typeof(RawJsonConverter))] can cause issues. For example:
```
string metadata = "this is my metadata";
PaymentRequest paymentRequest = new PaymentRequest() {
    Amount = new Amount(Currency.EUR, "100.00"),
    Description = "Description",
    RedirectUrl = this.DefaultRedirectUrl,
    Metadata = metadata
};
PaymentResponse result = await this._paymentClient.CreatePaymentAsync(paymentRequest);
Assert.AreEqual(metadata, result.Metadata);
```

The above test will now fail. The RawJsonConverter can't handle non-json data. 

Kind regards,
Vincent

**Comment by Viincenttt** [comment]
Hi pGuernseyPlayer,

I just wanted to let you know that this issue was also reported in #99. I have implemented a solution that should work for all metadata cases (string or json objects). A new NuGet package that includes this fix will be released soon. 

Kind regards,
Vincent

**Comment by Viincenttt** [comment]
The issue should now be resolved in version 2.0.3.0, which is available for download.
